src/lib/metrics.js

```diff
@@ -0,0 +1,99 @@
+// Track recent tabs woken to see whether they're later closed or focused
+let unseenWakeHistory = {};
+let seenWakeHistory = {};
+
+// BroadcastChannel for sending metrics pings to Test Pilot add-on
+let pingChannel = null;
+
+// Name of the metrics BroadcastChannel expected by the Test Pilot add-on
+const TESTPILOT_TELEMETRY_CHANNEL = 'testpilot-telemetry';
```

Comment:
There seems to be a lot of Test-Pilot-specific _and_ SnoozeTabs-specific stuff in here.  How easy would it be to switch this over to use Google Analytics if it ends up graduating into a Shield study?  Would it make sense to split the Test-Pilot-specific stuff out into a separate file that the SnoozeTabs-specific stuff could call, and then switch that file out for a Google-Analytics-specific implementation?